lib/ancestry/instance_methods.rb

```diff
@@ -288,16 +284,21 @@ def ancestry_callbacks_disabled?
 
   private
 
-    def cast_primary_key(key)
-      if [:string, :uuid, :text].include? primary_key_type
-        key
-      else
-        key.to_i
-      end
+    def parse_ancestry_column obj
+      cast_primary_keys(obj.to_s.split('/'))
+    end
+
+    def cast_primary_keys(keys)
+      skip_cast? ? keys : keys.map(&:to_i)
+    end
+
+    def skip_cast?
+      return @skip_cast if defined?(@skip_cast)
+      @skip_cast ||= [:string, :uuid, :text].include?(primary_key_type)
     end
 
     def primary_key_type
-      @primary_key_type ||= column_for_attribute(self.class.primary_key).type
+      column_for_attribute(self.class.primary_key).type
```

Comment:
Confused why you got rid of the memoization here.  Doesn't seem like it would help, but only hurt.

Reply:
this is only used in 1 place (to determine if we should skip keys) - so decided to only keep 1 in memory. although, it is probably pretty minor